test-data/stubgen/pybind11_mypy_demo/basics.pyi

```diff
@@ -5,8 +5,6 @@ PI: float
 
 class Point:
     class AngleUnit:
-        __doc__: ClassVar[str] = ...  # read-only
```

Comment:
Since stubs for special dunder attributes is not expected to be generated, we remove these from the test files.

Reply:
But we had tests explicitly adding them, so I'd like to make sure these aren't important for pybind11 users.

@sizmailov added these tests in #9903; could you comment on whether removing these attributes from the docs is desirable?

Reply:
I agree, annotation of most of dunder attributes makes a little sense and probably utilized only by few people if any, but I cannot speak for whole pybind11 community.
At the time I've added all the attributes to test just because it was generated by the stubgen and I saw no harm in leaving them.

It looks like dunder attributes hiding is orthogonal to the objectives of #12150.

Reply:
>It looks like dunder attributes hiding is orthogonal to the objectives of #12150.

I agree on this, I had added this change to be in-line with stub generation logic for python files where we ignore special dunder methods ([Ref](https://github.com/python/mypy/blob/eef316d65bf583b081f0101cdafbef3ecc353acc/mypy/semanal_enum.py#L23)).

We can also choose to not ignore dunder methods in `generate_c_property_stub`, in that case, we would have to update the below two tests since these will now generate stubs for dunder attributes: 

 - [test_generate_c_type_stub_variable_type_annotation ](https://github.com/python/mypy/blob/master/mypy/test/teststubgen.py#L678) 
`AssertionError: ['class C(KeyError):', '    __weakref__: Any'] != ['class C(KeyError): ...']`

 - [test_generate_c_type_inheritance](https://github.com/python/mypy/blob/master/mypy/test/teststubgen.py#L690) 
`AssertionError: ['class C:', '    x: ClassVar[int] = ...', '    __dict__: Any', '    __weakref__: Any'] != ['class C:', '    x: ClassVar[int] = ...']
`

@sizmailov Let me know if this sounds fine to you and I would then go ahead with the change.